pkgs/applications/networking/instant-messengers/cinny-desktop/default.nix

```diff
@@ -0,0 +1,51 @@
+{ stdenv
+, lib
+, dpkg
+, fetchurl
+, autoPatchelfHook
+, openssl
+, gnused
+, webkitgtk
+, makeWrapper
+, disableCompositing ? false # https://bugs.webkit.org/show_bug.cgi?id=180739
+}:
+
+stdenv.mkDerivation rec {
+
+  name = "cinny-desktop";
+  version = "2.0.4";
+
+  src = fetchurl {
+    url = "https://github.com/cinnyapp/cinny-desktop/releases/download/v${version}/cinny_${version}_amd64.deb";
+    sha256 = "sha256-udjLlYZTqOBTF9H7N6wJnIomFAeQvAczmEp3imE2IGY=";
+  };
+
+  nativeBuildInputs =
+    [ autoPatchelfHook dpkg gnused ]
+    ++ lib.optionals disableCompositing [ makeWrapper ];
+
+  buildInputs = [
+    openssl
+    webkitgtk
+  ];
+
+  unpackPhase = "true";
+
+  installPhase = ''
+    mkdir -p $out/bin $out/share
+    dpkg -x $src $out/tmp
+    sed -i 's/Exec=cinny/Exec=cinny-desktop/g' $out/tmp/usr/share/applications/cinny.desktop
+    cp $out/tmp/usr/bin/cinny $out/bin/cinny-desktop
```

Comment:
Please use `intall -Dm755` to install executable. It would also be preferable to use it to install other file where possible.

Reply:
Not sure if this comment is still valid for the latest version of the PR.
I've taken inspiration from other builds in Nixpkgs now.